adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -613,39 +635,38 @@ private boolean verifyAuthenticator(final AccountManager am) {
      */
     @Override
     public UserInfo[] getBrokerUsers() throws OperationCanceledException, AuthenticatorException, IOException {
-
+        final String methodName = ":getBrokerUsers";
         // Calling this on main thread will cause exception since this is
         // waiting on AccountManagerFuture
         if (Looper.myLooper() == Looper.getMainLooper()) {
             throw new IllegalArgumentException("Calling getBrokerUsers on main thread");
         }
 
-        Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
-        Bundle bundle = new Bundle();
+        final Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
```

Comment:
you should early return if accountList is of length 0, else line 647 is wrong by setting user info to true